### PR TITLE
Fix variation_of_information

### LINKED ISSUE #4856: Problem in variation of infromation, should we ignore 0 in pred?
## Description
I found there are problem in the calculation of variation of information.
I found that the codes of vi in skimage and cremi are also based on gala's code.
*gala: https://github.com/janelia-flyem/gala/blob/master/gala/evaluate.py
*cremi: https://github.com/cremi/cremi_python/blob/master/cremi/evaluation/voi.py
But there is a tiny question: Where and why to ignore 0 in VI?
If I don't ignore 0 in the prediction, I match [1, 1, 0] in pred to [1, 2, 3] in GT, and I get 2/3 * 1 + 1/3 * 0 = 2/3 for the VI.

Beside
For gala code,  they ignore 0 both in gt and pred (Default)
For cremi code, they only ignore 0 in gt, not in pred (Deafult)
For skimage, you did not ignore 0 in gt and pred (Default)
I made a simple comparison below, and i found that different setting will draw out different result. 
image

### PR DESCRIPTION
**Pull request by rfezzani**
## Description

Fixes #4856

The original bug is in the contingency table normalization.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Comment by jni** [comment]
Not sure why pooch is failing in *some* builds with:

> 2020-07-31T10:27:45.5728429Z E           ValueError: SHA256 hash of downloaded file (cells.tif) does not match the known hash: expected 2120cfe08e0396324793a10a905c9bbcb64b117215eb63b2c24b643e1600c8c9 but got ce67cfa98abd07c0ed4cedeac795ab118a65aa166d5ddbeb6fad756793c3bb4a. Deleted download for safety. The downloaded file may have been corrupted or the known hash may be outdated.

I guess we can ignore as not relevant to this PR!

**Comment by rfezzani** [comment]
Thank you @jni, sorry for changing the value of `ignore_labels` for `()` to `None`, but it really jumped to my eyes and I couldn't resist ^^

File: skimage/metrics/_contingency_table.py [hunk 1]
```diff
@@ -27,14 +28,12 @@ def contingency_table(im_true, im_test, *, ignore_labels=(), normalize=False):
...
-    im_test_r = im_test.ravel()
-    im_true_r = im_true.ravel()
-    ignored = np.zeros(im_true_r.shape, np.bool)
-    for label in ignore_labels:
-        ignored[im_true_r == label] = True
-    data = np.ones(im_true_r.shape)
-    data[ignored] = 0
+    if ignore_labels is None:
+        ignore_labels = []
+    im_test_r = im_test.reshape(-1)
```
Comment (emmanuelle) [inline skimage/metrics/_contingency_table.py:33]: just curious to know why you changed `im.ravel()` to `im.reshape(-1)`?

Comment (emmanuelle) [reply skimage/metrics/_contingency_table.py:33]: (it's just for my own curiosity, no problem with the proposal)

**Comment by emmanuelle** [comment]
Thanks @rfezzani, merging.

File: skimage/metrics/_contingency_table.py (hunk 1 above)
Comment (jni) [reply skimage/metrics/_contingency_table.py:33]: @emmanuelle The answer is in the .ravel docstring: .reshape is less likely to result in a copy. Which surprised me but after several reminders @rfezzani has finally hammered it into my brain.

Comment (emmanuelle) [reply skimage/metrics/_contingency_table.py:33]: now that you say it I'm almost sure I already asked @rfezzani this same question. Thanks for answering, we're doing group learning here :-).

**Pull request merged**